src/net_processing.cpp

```diff
@@ -914,6 +929,28 @@ void PeerManagerImpl::PushNodeVersion(CNode& pnode, int64_t nTime)
     }
 }
 
+size_t PeerManagerImpl::GetFloodingOutboundsCount(const CNode& skip_node) const
```

Comment:
instead of recalculating whenever we get a `SENDRECON` message, could this logic be significantly simplified by just storing a count that gets updated when nodes do something to change their state? the characteristics being checked are all established early in the connection (connection type, m_tx_relay struct, flood_to being set) 

I think this should remove the need for the "skip node" logic in this function, but if not, the node being processed can be checked and then subtract one if needed.